Update UG and DG (#114)
AY2223S1-CS2103T-T11-3/tp: docs/DeveloperGuide.md

  
    Use case resumes from step 2.
  
**Use case: UC17 - Mark a student as absent for a lesson**

Refer to <ins>UC16 - Mark a student as present for a lesson</ins> with the only difference being mark a student as absent instead of present.

**Use case: UC18 - Add note for a student in a lesson**

Preconditions: There are existing students and lessons in JARVIS.

1. Avenger performs <ins>view lessons(UC3)</ins>.
2. Avenger requests to add a note for a student in a lesson.
3. JARVIS adds the note for the student to the lesson.

Use case ends.

**Extensions:**
- 2a. JARVIS fails to understand request.
    - 2a1. JARVIS tells Avenger to make a request again.
  
      Use case resumes from step 2.
  
- 2b. Specified lesson not found in JARVIS.
    - 2b1. JARVIS informs Avenger that the lesson does not exist.
    - 2b2. JARVIS tells Avenger to make a request again.
  
      Use case resumes from step 2.

- 2c. Specified student not found in JARVIS.
    - 2c1. JARVIS informs Avenger that the student does not exist in the lesson.
    - 2c2. JARVIS tells Avenger to make a request again.

      Use case resumes from step 2.

**Use case: UC19 - Delete note for a student in a lesson**

> Refer to <ins>UC18 - Add note for a student in a lesson</ins> with the only difference being delete note instead of add note.

Reviewer: I think this use case might be slightly different because there is the possibility of the note index being invalid

> **Use case: UC20 - Add participation for a student in a lesson**

Reviewer: Maybe "Set participation" would be better phrasing here

Refer to <ins>UC18 - Add note for a student in a lesson</ins> with the only difference being adding participation instead of adding note.  
    
**Use case: UC21 - Delete a student**

Preconditions: There are existing students in JARVIS.

**MSS**

1. Avenger performs <ins>view students(UC1)</ins>.
2. Avenger requests to delete a student.
3. JARVIS deletes the student.

Use case ends.

**Extensions**

- 2a. JARVIS fails to understand request.
  - 2a1. JARVIS tells Avenger to make a request again.

    Use case resumes from step 2.
  
- 2b. Specified student not found in JARVIS.
  - 2b1. JARVIS informs Avenger that the student does not exist.

>   - 2b2. JARVIS 

Reviewer: Incomplete line

> **Use case: UC22 - Delete a task**

Reviewer: I think this is similar to UC21 - Delete a student

**Use case: UC18 - Add note for a student in a lesson**

Preconditions: There are existing students and lessons in JARVIS.

1. Avenger performs <ins>view lessons(UC3)</ins>.
2. Avenger requests to add a note for a student in a lesson.
3. JARVIS adds the note for the student to the lesson.

Use case ends.

**Extensions:**
- 2a. JARVIS fails to understand request.
    - 2a1. JARVIS tells Avenger to make a request again.
  
      Use case resumes from step 2.
  
- 2b. Specified lesson not found in JARVIS.
    - 2b1. JARVIS informs Avenger that the lesson does not exist.
    - 2b2. JARVIS tells Avenger to make a request again.
  
      Use case resumes from step 2.

- 2c. Specified student not found in JARVIS.
    - 2c1. JARVIS informs Avenger that the student does not exist in the lesson.
    - 2c2. JARVIS tells Avenger to make a request again.

      Use case resumes from step 2.

**Use case: UC19 - Delete note for a student in a lesson**

Preconditions: There are existing students, lessons and notes for a student in JARVIS.

1. Avenger performs <ins>view lessons(UC3)</ins>.
2. Avenger requests to delete a note from a student in a lesson.

> 3. JARVIS deletes the note from the student to the lesson.

Reviewer: ```suggestion
3. JARVIS deletes the note for the student in the lesson.
```
Edited to match the name of the Use Case

Use case ends.

**Extensions:**
- 2a. JARVIS fails to understand request.
    - 2a1. JARVIS tells Avenger to make a request again.

      Use case resumes from step 2.

- 2b. Specified lesson not found in JARVIS.
    - 2b1. JARVIS informs Avenger that the lesson does not exist.
    - 2b2. JARVIS tells Avenger to make a request again.

      Use case resumes from step 2.

- 2c. Specified student not found in JARVIS.
    - 2c1. JARVIS informs Avenger that the student does not exist in the lesson.
    - 2c2. JARVIS tells Avenger to make a request again.

      Use case resumes from step 2.
  
- 2d. Specified note not found in JARVIS.
  - 2d1. JARVIS informs Avenger that the note does not exist for the student in the lesson.

>   - 2d2. JARVIS tells Avenger to make a request again.

Reviewer: ```suggestion
  - 2d2. JARVIS tells Avenger to make a request again.
  
    Use case resumes from step 2.
```
